### PR TITLE
Create separate build configuration for staging

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
There are many CORS errors on https://staging.expensify.cash 

image

And it is because it is using the wrong value for the `EXPENSIFY_URL_CASH` environment variable.

### Fixed Issues
No issue, hot-fixing staging.

### Tests

Tested with local build, which is the best we can do for now.

### CODE REVIEW

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

**Comment by roryabraham** [comment]
Yes, I did run a local build:

```
  npm run build-staging

> expensify.cash@1.0.5-31 build-staging /Users/roryabraham/ReactNativeChat
> webpack --config config/webpack/webpack.staging.js

Hash: 37b365ec39dd395deec9
Version: webpack 4.44.2
Time: 45396ms
Built at: 03/24/2021 5:19:23 PM
                                                     Asset       Size  Chunks                                Chunk Names
                      0c783d1b59ec31130b1c020427c3d55a.png   5.71 KiB          [emitted] [immutable]         
                        app-37b365ec39dd395deec9.bundle.js   2.03 MiB       0  [emitted] [immutable]  [big]  app
                    app-37b365ec39dd395deec9.bundle.js.map   6.67 MiB       0  [emitted] [dev]               app
                      b200a37389bc3691eb4cbb0e800e83a7.png  116 bytes          [emitted] [immutable]         
                      bfc818e9327a5fb2240124e0da8dd205.png     39 KiB          [emitted] [immutable]         
                      c733a880bd6d3d1f0f38ef1608fb8a32.png  207 bytes          [emitted] [immutable]         
                                      cmaps/78-EUC-H.bcmap   2.35 KiB          [emitted]                     
                                      cmaps/78-EUC-V.bcmap  173 bytes          [emitted]                     
                                          cmaps/78-H.bcmap   2.32 KiB          [emitted]                     
                                     cmaps/78-RKSJ-H.bcmap   2.34 KiB          [emitted]                     
                                     cmaps/78-RKSJ-V.bcmap  173 bytes          [emitted]                     
                                          cmaps/78-V.bcmap  169 bytes          [emitted]                     
                                   cmaps/78ms-RKSJ-H.bcmap   2.59 KiB          [emitted]                     
                                   cmaps/78ms-RKSJ-V.bcmap  290 bytes          [emitted]                     
                                   cmaps/83pv-RKSJ-H.bcmap  905 bytes          [emitted]                     
                                   cmaps/90ms-RKSJ-H.bcmap  721 bytes          [emitted]                     
                                   cmaps/90ms-RKSJ-V.bcmap  290 bytes          [emitted]                     
                                  cmaps/90msp-RKSJ-H.bcmap  715 bytes          [emitted]                     
                                  cmaps/90msp-RKSJ-V.bcmap  291 bytes          [emitted]                     
                                   cmaps/90pv-RKSJ-H.bcmap  982 bytes          [emitted]                     
                                   cmaps/90pv-RKSJ-V.bcmap  260 bytes          [emitted]                     
                                         cmaps/Add-H.bcmap   2.36 KiB          [emitted]                     
                                    cmaps/Add-RKSJ-H.bcmap   2.36 KiB          [emitted]                     
                                    cmaps/Add-RKSJ-V.bcmap  287 bytes          [emitted]                     
                                         cmaps/Add-V.bcmap  282 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-0.bcmap  317 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-1.bcmap  371 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-2.bcmap  376 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-3.bcmap  401 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-4.bcmap  405 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-5.bcmap  406 bytes          [emitted]                     
                                  cmaps/Adobe-CNS1-6.bcmap  406 bytes          [emitted]                     
                               cmaps/Adobe-CNS1-UCS2.bcmap   40.2 KiB          [emitted]                     
                                   cmaps/Adobe-GB1-0.bcmap  217 bytes          [emitted]                     
                                   cmaps/Adobe-GB1-1.bcmap  250 bytes          [emitted]                     
                                   cmaps/Adobe-GB1-2.bcmap  465 bytes          [emitted]                     
                                   cmaps/Adobe-GB1-3.bcmap  470 bytes          [emitted]                     
                                   cmaps/Adobe-GB1-4.bcmap  601 bytes          [emitted]                     
                                   cmaps/Adobe-GB1-5.bcmap  625 bytes          [emitted]                     
                                cmaps/Adobe-GB1-UCS2.bcmap   33.2 KiB          [emitted]                     
                                cmaps/Adobe-Japan1-0.bcmap  225 bytes          [emitted]                     
                                cmaps/Adobe-Japan1-1.bcmap  226 bytes          [emitted]                     
                                cmaps/Adobe-Japan1-2.bcmap  233 bytes          [emitted]                     
                                cmaps/Adobe-Japan1-3.bcmap  242 bytes          [emitted]                     
                                cmaps/Adobe-Japan1-4.bcmap  337 bytes          [emitted]                     
                                cmaps/Adobe-Japan1-5.bcmap  430 bytes          [emitted]                     
                                cmaps/Adobe-Japan1-6.bcmap  485 bytes          [emitted]                     
                             cmaps/Adobe-Japan1-UCS2.bcmap     40 KiB          [emitted]                     
                                cmaps/Adobe-Korea1-0.bcmap  241 bytes          [emitted]                     
                                cmaps/Adobe-Korea1-1.bcmap  386 bytes          [emitted]                     
                                cmaps/Adobe-Korea1-2.bcmap  391 bytes          [emitted]                     
                             cmaps/Adobe-Korea1-UCS2.bcmap   22.7 KiB          [emitted]                     
                                          cmaps/B5-H.bcmap   1.06 KiB          [emitted]                     
                                          cmaps/B5-V.bcmap  142 bytes          [emitted]                     
                                        cmaps/B5pc-H.bcmap   1.07 KiB          [emitted]                     
                                        cmaps/B5pc-V.bcmap  144 bytes          [emitted]                     
                                     cmaps/CNS-EUC-H.bcmap   1.74 KiB          [emitted]                     
                                     cmaps/CNS-EUC-V.bcmap   1.88 KiB          [emitted]                     
                                        cmaps/CNS1-H.bcmap  706 bytes          [emitted]                     
                                        cmaps/CNS1-V.bcmap  143 bytes          [emitted]                     
                                        cmaps/CNS2-H.bcmap  504 bytes          [emitted]                     
                                        cmaps/CNS2-V.bcmap   93 bytes          [emitted]                     
                                     cmaps/ETHK-B5-H.bcmap   4.32 KiB          [emitted]                     
                                     cmaps/ETHK-B5-V.bcmap  158 bytes          [emitted]                     
                                     cmaps/ETen-B5-H.bcmap    1.1 KiB          [emitted]                     
                                     cmaps/ETen-B5-V.bcmap  158 bytes          [emitted]                     
                                   cmaps/ETenms-B5-H.bcmap  101 bytes          [emitted]                     
                                   cmaps/ETenms-B5-V.bcmap  172 bytes          [emitted]                     
                                         cmaps/EUC-H.bcmap  578 bytes          [emitted]                     
                                         cmaps/EUC-V.bcmap  170 bytes          [emitted]                     
                                         cmaps/Ext-H.bcmap   2.48 KiB          [emitted]                     
                                    cmaps/Ext-RKSJ-H.bcmap   2.48 KiB          [emitted]                     
                                    cmaps/Ext-RKSJ-V.bcmap  218 bytes          [emitted]                     
                                         cmaps/Ext-V.bcmap  215 bytes          [emitted]                     
                                      cmaps/GB-EUC-H.bcmap  549 bytes          [emitted]                     
                                      cmaps/GB-EUC-V.bcmap  179 bytes          [emitted]                     
                                          cmaps/GB-H.bcmap  528 bytes          [emitted]                     
                                          cmaps/GB-V.bcmap  175 bytes          [emitted]                     
                                     cmaps/GBK-EUC-H.bcmap   14.3 KiB          [emitted]                     
                                     cmaps/GBK-EUC-V.bcmap  180 bytes          [emitted]                     
                                       cmaps/GBK2K-H.bcmap   19.2 KiB          [emitted]                     
                                       cmaps/GBK2K-V.bcmap  219 bytes          [emitted]                     
                                    cmaps/GBKp-EUC-H.bcmap   14.3 KiB          [emitted]                     
                                    cmaps/GBKp-EUC-V.bcmap  181 bytes          [emitted]                     
                                     cmaps/GBT-EUC-H.bcmap   7.12 KiB          [emitted]                     
                                     cmaps/GBT-EUC-V.bcmap  180 bytes          [emitted]                     
                                         cmaps/GBT-H.bcmap    7.1 KiB          [emitted]                     
                                         cmaps/GBT-V.bcmap  176 bytes          [emitted]                     
                                   cmaps/GBTpc-EUC-H.bcmap   7.13 KiB          [emitted]                     
                                   cmaps/GBTpc-EUC-V.bcmap  182 bytes          [emitted]                     
                                    cmaps/GBpc-EUC-H.bcmap  557 bytes          [emitted]                     
                                    cmaps/GBpc-EUC-V.bcmap  181 bytes          [emitted]                     
                                             cmaps/H.bcmap  553 bytes          [emitted]                     
                                    cmaps/HKdla-B5-H.bcmap   2.59 KiB          [emitted]                     
                                    cmaps/HKdla-B5-V.bcmap  148 bytes          [emitted]                     
                                    cmaps/HKdlb-B5-H.bcmap   2.36 KiB          [emitted]                     
                                    cmaps/HKdlb-B5-V.bcmap  148 bytes          [emitted]                     
                                   cmaps/HKgccs-B5-H.bcmap   2.24 KiB          [emitted]                     
                                   cmaps/HKgccs-B5-V.bcmap  149 bytes          [emitted]                     
                                   cmaps/HKm314-B5-H.bcmap   1.73 KiB          [emitted]                     
                                   cmaps/HKm314-B5-V.bcmap  149 bytes          [emitted]                     
                                   cmaps/HKm471-B5-H.bcmap   2.12 KiB          [emitted]                     
                                   cmaps/HKm471-B5-V.bcmap  149 bytes          [emitted]                     
                                    cmaps/HKscs-B5-H.bcmap   4.33 KiB          [emitted]                     
                                    cmaps/HKscs-B5-V.bcmap  159 bytes          [emitted]                     
                                       cmaps/Hankaku.bcmap  132 bytes          [emitted]                     
                                      cmaps/Hiragana.bcmap  124 bytes          [emitted]                     
                                     cmaps/KSC-EUC-H.bcmap    1.8 KiB          [emitted]                     
                                     cmaps/KSC-EUC-V.bcmap  164 bytes          [emitted]                     
                                         cmaps/KSC-H.bcmap   1.79 KiB          [emitted]                     
                                   cmaps/KSC-Johab-H.bcmap   16.4 KiB          [emitted]                     
                                   cmaps/KSC-Johab-V.bcmap  166 bytes          [emitted]                     
                                         cmaps/KSC-V.bcmap  160 bytes          [emitted]                     
                                   cmaps/KSCms-UHC-H.bcmap   2.72 KiB          [emitted]                     
                                cmaps/KSCms-UHC-HW-H.bcmap   2.72 KiB          [emitted]                     
                                cmaps/KSCms-UHC-HW-V.bcmap  169 bytes          [emitted]                     
                                   cmaps/KSCms-UHC-V.bcmap  166 bytes          [emitted]                     
                                   cmaps/KSCpc-EUC-H.bcmap   1.98 KiB          [emitted]                     
                                   cmaps/KSCpc-EUC-V.bcmap  166 bytes          [emitted]                     
                                      cmaps/Katakana.bcmap  100 bytes          [emitted]                     
                                             cmaps/LICENSE   2.03 KiB          [emitted]                     
                                         cmaps/NWP-H.bcmap    2.7 KiB          [emitted]                     
                                         cmaps/NWP-V.bcmap  252 bytes          [emitted]                     
                                        cmaps/RKSJ-H.bcmap  534 bytes          [emitted]                     
                                        cmaps/RKSJ-V.bcmap  170 bytes          [emitted]                     
                                         cmaps/Roman.bcmap   96 bytes          [emitted]                     
                                 cmaps/UniCNS-UCS2-H.bcmap   47.1 KiB          [emitted]                     
                                 cmaps/UniCNS-UCS2-V.bcmap  156 bytes          [emitted]                     
                                cmaps/UniCNS-UTF16-H.bcmap   49.2 KiB          [emitted]                     
                                cmaps/UniCNS-UTF16-V.bcmap  156 bytes          [emitted]                     
                                cmaps/UniCNS-UTF32-H.bcmap   51.4 KiB          [emitted]                     
                                cmaps/UniCNS-UTF32-V.bcmap  160 bytes          [emitted]                     
                                 cmaps/UniCNS-UTF8-H.bcmap   52.4 KiB          [emitted]                     
                                 cmaps/UniCNS-UTF8-V.bcmap  157 bytes          [emitted]                     
                                  cmaps/UniGB-UCS2-H.bcmap   42.3 KiB          [emitted]                     
                                  cmaps/UniGB-UCS2-V.bcmap  193 bytes          [emitted]                     
                                 cmaps/UniGB-UTF16-H.bcmap   43.1 KiB          [emitted]                     
                                 cmaps/UniGB-UTF16-V.bcmap  178 bytes          [emitted]                     
                                 cmaps/UniGB-UTF32-H.bcmap   44.7 KiB          [emitted]                     
                                 cmaps/UniGB-UTF32-V.bcmap  182 bytes          [emitted]                     
                                  cmaps/UniGB-UTF8-H.bcmap   45.7 KiB          [emitted]                     
                                  cmaps/UniGB-UTF8-V.bcmap  181 bytes          [emitted]                     
                                 cmaps/UniJIS-UCS2-H.bcmap   24.8 KiB          [emitted]                     
                              cmaps/UniJIS-UCS2-HW-H.bcmap  119 bytes          [emitted]                     
                              cmaps/UniJIS-UCS2-HW-V.bcmap  680 bytes          [emitted]                     
                                 cmaps/UniJIS-UCS2-V.bcmap  664 bytes          [emitted]                     
                                cmaps/UniJIS-UTF16-H.bcmap   38.5 KiB          [emitted]                     
                                cmaps/UniJIS-UTF16-V.bcmap  643 bytes          [emitted]                     
                                cmaps/UniJIS-UTF32-H.bcmap   39.6 KiB          [emitted]                     
                                cmaps/UniJIS-UTF32-V.bcmap  677 bytes          [emitted]                     
                                 cmaps/UniJIS-UTF8-H.bcmap   40.7 KiB          [emitted]                     
                                 cmaps/UniJIS-UTF8-V.bcmap  678 bytes          [emitted]                     
                            cmaps/UniJIS2004-UTF16-H.bcmap   38.6 KiB          [emitted]                     
                            cmaps/UniJIS2004-UTF16-V.bcmap  647 bytes          [emitted]                     
                            cmaps/UniJIS2004-UTF32-H.bcmap   39.7 KiB          [emitted]                     
                            cmaps/UniJIS2004-UTF32-V.bcmap  681 bytes          [emitted]                     
                             cmaps/UniJIS2004-UTF8-H.bcmap   40.8 KiB          [emitted]                     
                             cmaps/UniJIS2004-UTF8-V.bcmap  682 bytes          [emitted]                     
                           cmaps/UniJISPro-UCS2-HW-V.bcmap  705 bytes          [emitted]                     
                              cmaps/UniJISPro-UCS2-V.bcmap  689 bytes          [emitted]                     
                              cmaps/UniJISPro-UTF8-V.bcmap  726 bytes          [emitted]                     
                           cmaps/UniJISX0213-UTF32-H.bcmap   39.6 KiB          [emitted]                     
                           cmaps/UniJISX0213-UTF32-V.bcmap  684 bytes          [emitted]                     
                       cmaps/UniJISX02132004-UTF32-H.bcmap   39.7 KiB          [emitted]                     
                       cmaps/UniJISX02132004-UTF32-V.bcmap  688 bytes          [emitted]                     
                                  cmaps/UniKS-UCS2-H.bcmap   25.2 KiB          [emitted]                     
                                  cmaps/UniKS-UCS2-V.bcmap  178 bytes          [emitted]                     
                                 cmaps/UniKS-UTF16-H.bcmap   25.7 KiB          [emitted]                     
                                 cmaps/UniKS-UTF16-V.bcmap  164 bytes          [emitted]                     
                                 cmaps/UniKS-UTF32-H.bcmap   25.8 KiB          [emitted]                     
                                 cmaps/UniKS-UTF32-V.bcmap  168 bytes          [emitted]                     
                                  cmaps/UniKS-UTF8-H.bcmap   27.1 KiB          [emitted]                     
                                  cmaps/UniKS-UTF8-V.bcmap  169 bytes          [emitted]                     
                                             cmaps/V.bcmap  166 bytes          [emitted]                     
                                     cmaps/WP-Symbol.bcmap  179 bytes          [emitted]                     
                                   css/AnnotationLayer.css   5.42 KiB          [emitted]                     
                                             css/fonts.css   3.89 KiB          [emitted]                     
                                               css/pdf.css  683 bytes          [emitted]                     
                      fa3a0dc491504bbdaa054914c7a76f6d.png     43 KiB          [emitted] [immutable]         
                      fae3c934a7353e1defbb00343509b3aa.png  913 bytes          [emitted] [immutable]         
                                        favicon-unread.png   6.26 KiB          [emitted]                     
                                               favicon.png   5.71 KiB          [emitted]                     
                                         images/shadow.png  269 bytes          [emitted]                     
                                                index.html   1.42 KiB          [emitted]                     
                                      og-preview-image.png   16.5 KiB          [emitted]                     
    pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js    624 KiB          [emitted] [immutable]  [big]  
pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js.map   2.23 MiB          [emitted] [dev]               
Entrypoint app [big] = app-37b365ec39dd395deec9.bundle.js app-37b365ec39dd395deec9.bundle.js.map
  [5] ./node_modules/underscore/modules/index-all.js + 152 modules 76.3 KiB {0} [built]
      |    153 modules
 [12] ./node_modules/react-native-web/dist/exports/StyleSheet/index.js + 1 modules 2.35 KiB {0} [built]
      |    2 modules
 [15] ./node_modules/react-native-web/dist/exports/Animated/index.js + 13 modules 63.8 KiB {0} [built]
      |    14 modules
 [18] ./node_modules/react-native-web/dist/index.js + 26 modules 44.4 KiB {0} [built]
      |    27 modules
 [43] ./node_modules/react-native-web/dist/exports/Image/index.js + 2 modules 17.5 KiB {0} [built]
      |    3 modules
 [58] ./node_modules/react-native-web/dist/exports/ScrollView/index.js + 2 modules 37.5 KiB {0} [built]
      |    3 modules
 [72] ./node_modules/react-native-gesture-handler/RNGestureHandlerModule.web.js + 17 modules 163 KiB {0} [built]
      |    18 modules
 [75] ./node_modules/react-native-web/dist/exports/FlatList/index.js + 2 modules 18.8 KiB {0} [built]
      |    3 modules
 [95] ./node_modules/react-native-web/dist/exports/SectionList/index.js + 2 modules 24.4 KiB {0} [built]
      |    3 modules
[111] ./node_modules/react-native-web/dist/exports/AppRegistry/index.js + 2 modules 8.33 KiB {0} [built]
      |    3 modules
[116] ./package.json 7.23 KiB {0} [built]
[119] ./node_modules/react-native-web/dist/exports/StyleSheet/styleResolver.js + 13 modules 53.6 KiB {0} [built]
      |    14 modules
[366] ./desktop/ELECTRON_EVENTS.js 199 bytes {0} [built]
[368] ./src/libs/checkForUpdates.js 605 bytes {0} [built]
[672] ./index.js + 507 modules 1.56 MiB {0} [not cacheable] [built]
      | ./index.js 337 bytes [built]
      | ./src/CONFIG.js 2.25 KiB [built]
      | ./src/App.js 442 bytes [built]
      | ./src/setup/index.website.js 1.83 KiB [built]
      | ./src/libs/getPlatform/index.website.js 57 bytes [built]
      | ./src/libs/HttpUtils.js 1.8 KiB [built]
      | ./src/libs/Visibility/index.js 762 bytes [built]
      | ./src/Expensify.js 5.2 KiB [built]
      | ./src/libs/listenToStorageEvents/index.js 533 bytes [built]
      | ./src/ONYXKEYS.js 2.08 KiB [built]
      | ./src/libs/ActiveClientManager/index.js 830 bytes [built]
      | ./src/styles/styles.js 27.2 KiB [built]
      | ./src/libs/Notification/PushNotification/index.js 328 bytes [built]
      | ./src/libs/Log.js 1.16 KiB [built]
      | ./src/libs/Navigation/NavigationRoot.js 5.67 KiB [built]
      |     + 493 hidden modules
    + 666 hidden modules

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js (624 KiB)
  app-37b365ec39dd395deec9.bundle.js (2.03 MiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  app (2.03 MiB)
      app-37b365ec39dd395deec9.bundle.js


WARNING in webpack performance recommendations: 
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
Child HtmlWebpackCompiler:
     1 asset
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./web/index.html 1.78 KiB {0} [built]
Child worker-loader node_modules/react-pdf/dist/esm/pdf.worker.entry.js:
                                                         Asset      Size  Chunks                                Chunk Names
        pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js   624 KiB       0  [emitted] [immutable]  [big]  pdf.worker.entry
    pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js.map  2.23 MiB       0  [emitted] [dev]               pdf.worker.entry
    Entrypoint pdf.worker.entry = pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js pdf.worker.entry-471d04f41e48dccf3b68.bundle.worker.js.map
       3 modules

```

No errors.

**Comment by botify** [comment]
npm has a `package.json` file and a `package-lock.json` file. It seems you updated one without the other, which is usually a sign of a mistake. If you are updating a package make sure that you update the version in package.json then run `npm install`

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

File: config/webpack/webpack.staging.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const path = require('path');
+const webpack = require('webpack');
+const {merge} = require('webpack-merge');
+const {BundleAnalyzerPlugin} = require('webpack-bundle-analyzer');
+const dotenv = require('dotenv');
+const common = require('./webpack.common.js');
+
+const env = dotenv.config({path: path.resolve(__dirname, '../../.env.staging')}).parsed;
```
Comment (AndrewGable) [inline config/webpack/webpack.staging.js:8]: Is this the only difference between prod/staging? Is there a way to DRY this up more?

Comment (roryabraham) [reply config/webpack/webpack.staging.js:7]: DRYed up a bit 👍

**Review by AndrewGable — APPROVED** [review]

**Review by MariaHCD — APPROVED** [review]

**Comment by roryabraham** [comment]
Nice, this seemed to have worked 🎉

**Pull request merged**